backend/apps/rag/main.py

```diff
@@ -118,6 +118,10 @@ def store_data_in_vector_db(data, collection_name) -> bool:
     metadatas = [doc.metadata for doc in docs]
 
     try:
+        for collection in CHROMA_CLIENT.list_collections():
```

Comment:
Shouldn't we make it so that it only does this for web contents?

Reply:
Even when content does not change (file side of things), data is still fetched with loader.load(), split into chunks and it bails at the final step as collection already exists.

Still, calculating vectors is a considerable part of whole operation, so I guess you are right, it might be better to do this overwrite only for web side.

Let me figure out how to update a pull req :)